carto/auth.py

```diff
@@ -0,0 +1,106 @@
+"""
+Module for authenticated access to CARTO's APIs
+
+.. module:: carto.auth
+   :platform: Unix, Windows
+   :synopsis: Module for authenticated access to CARTO's APIs
+
+.. moduleauthor:: Daniel Carrion <daniel@carto.com>
+.. moduleauthor:: Alberto Romeu <alrocar@carto.com>
+
+
+"""
+
+import re
+import sys
+import warnings
+
+from pyrestcli.auth import BaseAuthClient
+
+from .exceptions import CartoException
+
+if sys.version_info >= (3, 0):
+    from urllib.parse import urlparse
+else:
+    from urlparse import urlparse
+
+
+class APIKeyAuthClient(BaseAuthClient):
+    """
+    This class provides you with authenticated access to CARTO's APIs using
+    your API key.
+
+    You can find your API key by clicking on the API key section of the user
+    dropdown menu
+    """
+    def __init__(self, base_url, api_key, organization=None, session=None):
```

Comment:
Let's keep it as `base_url` since changing to `domain` may be confusing for clients not using the cloud version of CARTO.

Thanks!